packages/firestore/test/unit/local/query_cache.test.ts

```diff
@@ -54,22 +51,22 @@ describe('IndexedDbQueryCache', () => {
     return;
   }
 
-  beforeEach(() => {
-    return persistenceHelpers.testIndexedDbPersistence().then(p => {
-      persistence = p;
-    });
+  let persistencePromise: Promise<Persistence<IndexedDbTransaction>>;
+  beforeEach(async () => {
+    persistencePromise = persistenceHelpers.testIndexedDbPersistence();
   });
 
-  afterEach(() => persistence.shutdown(/* deleteData= */ true));
-
-  genericQueryCacheTests();
+  genericQueryCacheTests(() => persistencePromise);
```

Comment:
The previous implementation used a different `persistence` object for every test case. Can you verify that this is still the case?

Reply:
Yes, the promise is recreated each time `beforeEach` runs, so each test run gets its own `persistence` instance.